timelock-impl/src/main/java/com/palantir/atlasdb/timelock/lock/BlockingTimeLimitedLockService.java

```diff
@@ -227,20 +246,26 @@ private static void logInterrupted(LockRequestSpecification specification, Inter
     }
 
     private BlockingTimeoutException logAndHandleTimeout(LockRequestSpecification specification) {
-        final String logMessage =
-                "Lock service timed out after {} milliseconds" + " when servicing {} for client \"{}\"";
-        log.info(
-                logMessage,
-                SafeArg.of("timeoutDurationMillis", blockingTimeLimitMillis),
-                SafeArg.of("method", specification.method()),
-                SafeArg.of("client", specification.client()),
-                UnsafeArg.of("lockRequest", specification.lockRequest()));
+        final String logMessage = "Lock service timed out after {} milliseconds when servicing {} for client \"{}\"";
+
+        List<Arg<?>> logArgs = new ArrayList<>();
+        logArgs.add(SafeArg.of("timeoutDurationMillis", blockingTimeLimitMillis));
+        logArgs.add(SafeArg.of("method", specification.method()));
+        logArgs.add(SafeArg.of("client", specification.client()));
+
+        if (threadInfoSnapshotManager.isRecordingThreadInfo()) {
```

Comment:
Consider just having threadInfoSnapshotManager just returning an unsafe arg with an optional snapshot, and have the unsafe arg name be suffixed with "ifEnabled", then you don't need to do the isRecordingThreadInfo check here (instead in the snapshotManager), and you don't need to have a list of args then convert to an array, instead just having each arg be on the log line, as it was before. You can do this for all the places you've added the logging arg